chainstate/src/detail/transaction_verifier/error.rs

```diff
@@ -86,6 +97,8 @@ pub enum ConnectTransactionError {
     SerializationInvariantError(Id<Block>),
     #[error("Timelock rules violated")]
     TimeLockViolation,
+    #[error("Utxo invariant broken: `{0}`")]
+    UtxoInvariantBroken(String),
```

Comment:
Please, no string errors in this library. Every error has to have a different type so that it can be handled by a receiving library. Even though it's an invariant error.